src/core/public/core_app/status/components/status_table.tsx

```diff
@@ -6,50 +6,115 @@
-    }),
-    render: (state: FormattedStatus['state']) => <span>{state.message}</span>,
-  },
-];
+const expandLabel = i18n.translate('core.statusPage.statusTable.columns.expandRow.expandLabel', {
+  defaultMessage: 'Expand',
+});
+
+const collapseLabel = i18n.translate(
+  'core.statusPage.statusTable.columns.expandRow.collapseLabel',
+  { defaultMessage: 'Collapse' }
+);
 
 export const StatusTable: FunctionComponent<StatusTableProps> = ({ statuses }) => {
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<
+    Record<string, ReactElement>
+  >({});
   if (!statuses) {
     return null;
   }
+
+  const toggleDetails = (item: FormattedStatus) => {
+    const newRowMap = { ...itemIdToExpandedRowMap };
+    if (itemIdToExpandedRowMap[item.id]) {
+      delete newRowMap[item.id];
+    } else {
+      newRowMap[item.id] = <StatusExpandedRow status={item} />;
+    }
+    setItemIdToExpandedRowMap(newRowMap);
+  };
+
+  const tableColumns: Array<EuiBasicTableColumn<FormattedStatus>> = [
+    {
+      field: 'state',
+      name: i18n.translate('core.statusPage.statusTable.columns.statusHeader', {
+        defaultMessage: 'Status',
+      }),
+      render: (state: FormattedStatus['state']) => (
+        <EuiIcon type="dot" aria-hidden color={state.uiColor} title={state.title} />
+      ),
+      width: '100px',
+      align: 'center' as const,
+      sortable: (row: FormattedStatus) => getLevelSortValue(row),
+    },
+    {
+      field: 'id',
+      name: i18n.translate('core.statusPage.statusTable.columns.idHeader', {
+        defaultMessage: 'ID',
+      }),
+      sortable: true,
+    },
+    {
+      field: 'state',
+      name: i18n.translate('core.statusPage.statusTable.columns.statusSummaryHeader', {
+        defaultMessage: 'Status summary',
+      }),
+      render: (state: FormattedStatus['state']) => <span>{state.message}</span>,
+    },
+    {
+      name: (
+        <EuiScreenReaderOnly>
+          <FormattedMessage
+            id="core.statusPage.statusTable.columns.expandRowHeader"
+            defaultMessage="Expand row"
+          />
+        </EuiScreenReaderOnly>
+      ),
+      align: 'right',
+      width: '40px',
+      isExpander: true,
+      render: (item: FormattedStatus) => (
+        <EuiButtonIcon
+          onClick={() => toggleDetails(item)}
+          aria-label={itemIdToExpandedRowMap[item.id] ? collapseLabel : expandLabel}
+          iconType={itemIdToExpandedRowMap[item.id] ? 'arrowUp' : 'arrowDown'}
+        />
+      ),
+    },
+  ];
+
   return (
-    <EuiBasicTable<FormattedStatus>
+    <EuiInMemoryTable<FormattedStatus>
       columns={tableColumns}
+      itemId={(item) => item.id}
       items={statuses}
+      isExpandable={true}
+      itemIdToExpandedRowMap={itemIdToExpandedRowMap}
       rowProps={({ state }) => ({
         className: `status-table-row-${state.uiColor}`,
       })}
+      sorting={{
+        sort: {
+          direction: 'asc',
+          field: 'state',
```

Comment:
It would be nice to sort on other columns too (e.g. plugin/service name) but can be handled in a follow up.

Reply:
I enabled sorting on the `status` and `id` columns (but this is done from the column definitions, so upper in the file)